File: examples/python/loop-shapes/LoopShapes.runar.py
from runar import SmartContract, Bigint, public, assert_


class LoopShapes(SmartContract):
    # LoopShapes — Python port. A NON-ZERO loop start (R-102).
    target: Bigint

    def __init__(self, target: Bigint):
        super().__init__(target)
        self.target = target

    @public
    def verify(self, seed: Bigint):
        acc: Bigint = seed
        for i in range(3, 7):
            acc = acc + i
        assert_(acc == self.target)

File: examples/ruby/loop-shapes/LoopShapes.runar.rb
require 'runar'

# LoopShapes -- Ruby port. A NON-ZERO loop start, ascending (R-102).
class LoopShapes < Runar::SmartContract
  prop :target, Bigint

  def initialize(target)
    super(target)
    @target = target
  end

  runar_public seed: Bigint
  def verify(seed)
    acc = seed
    for i in 3...7
      acc = acc + i
    end
    assert acc == @target
  end
end
